### PR TITLE
Implementations of type.all and type.convertAll

### LINKED ISSUE #31: Introduce a type.all
It would be nice to be able to convert a Type, and all of its properties, without creating a whole new type and changing all of its keys. For example say you had a type like:

```js
class Person extends ObservableObject {
  static props = {
    first: String,
    last: String,
    age: Number
  }
}
```

In an environment where you only have string values, you would want `age` to be converted. 

----

This proposal is for a `type.all` that creates a new TypeObject with its own `getSchema` symbol. This symbol will change all keys. It takes a type function as its first argument.

In the above example you would do:

```js
const ConvertingPerson = type.all(type.convert, Person);
```

This gives flexibility so that `type.all(type.check)` is also possible, if for some reason you needed that sort of behavior.

Since converting is the most common need we've seen I also propose a `type.convertAll(Type)` sugar function.

### PR DESCRIPTION
**Pull request by matthewp**
This adds implementations of `type.all` and `type.convertAll` as described in #31. This closes #31 

Example usage:

```js
import { ObservableObject, type } from "can";

class Person extends ObservableObject {
  static props = {
    age: Number
  };
}

let person = new Person({ age: "13" }); // throws

const ConvertingPerson = type.convertAll(Person);

person = new ConvertingPerson({ age: "13" });
console.log(person.age); // 13
```

Docs are included.

### CODE REVIEW

File: doc/all.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+@function can-type/all all
+@parent can-type/methods 7
+@description Derive a new type that changes the types of properties using a converter function.
+
+@signature `type.all(converter, Type)`
+
+  Given a converter function, which should be one of [can-type/check], [can-type/maybe], [can-type/convert], or [can-type/maybeConvert], create a new type that inherits the same properties of `Type`, but runs the converter function on them and returns a [can-type.typeobject].
```
Comment (phillipskevin) [inline doc/all.md]: Can we put
> which should be one of [can-type/check], [can-type/maybe], [can-type/convert], or [can-type/maybeConvert]

...in parentheses or in a separate sentence? I think the commas get a little confusing this way.

Also, I think
> but runs the converter function on them

is a little hard to follow.

I would phrase this something like

> Create a new type that inherits the same properties of `Type`, but with each property run through the `converter` function. The `converter` function should be one of [can-type/check], [can-type/maybe], [can-type/convert], or [can-type/maybeConvert]

File: can-type.js [hunk 2]
```diff
@@ -219,6 +219,34 @@ var Any = canReflect.assignSymbols({}, {
 	"can.isMember": function() { return true; }
 });
 
+function all(typeFn, Type) {
+	var typeObject = typeFn(Type);
```
Comment (phillipskevin) [inline can-type.js]: Is there a reason the `Type` needs to be passed through the converter also? (there probably is, just want to be sure)

File: doc/convertAll.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+@function can-type/convertAll convertAll
+@parent can-type/methods 8
+@description Derive a new type that makes every property of Type convert, eliminating strict type checking.
+
+@signature `type.convertAll(Type)`
+
+  Given a `Type`, creates a [can-type.typeobject] where each property on the Type is converted, rather than being strictly checked.
```
Comment (phillipskevin) [inline doc/convertAll.md]: Is showing the `typeobject` as the type of the `@return` below enough? I think it makes it more confusing to mention it here also. I think saying something like the `type.all` description makes it more clear:

> Create a new type that inherits the same properties of Type, but where each property on the Type is converted, rather than being strictly checked.

File: can-type.js (hunk 2 above)
Comment (matthewp) [reply can-type.js]: It's possible that this is not needed, let me check...

Comment (matthewp) [reply can-type.js]: Oh, we do want the isMemberSymbol. It might be better to make this return a typeObject and not a constructor, that way people don't attempt to use `instanceof` or something.

File: doc/all.md (hunk 1 above)
Comment (matthewp) [reply doc/all.md]: Sounds good!

File: can-type.js (hunk 2 above)
Comment (matthewp) [reply can-type.js]: After thinking about it, I left this alone because we want to inherit the `strictTypeOf` symbol and anything else that is on the typeObject. Because this is a constructor function it will inherit those things.  Type also might be a `typeObject` itself.

File: doc/convertAll.md (hunk 3 above)
Comment (matthewp) [reply doc/convertAll.md]: I did this.

File: doc/all.md (hunk 1 above)
Comment (matthewp) [reply doc/all.md]: Did this.
